server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -171,6 +171,12 @@ public ObjectMapper build(MapperBuilderContext context) {
     }
 
     public static class TypeParser implements Mapper.TypeParser {
+
+        @Override
+        public boolean supportsVersion(Version indexCreatedVersion) {
+            return true;
```

Comment:
here too, I wonder what this means. Does this get called, or does this simply mean "objects are always supported regardless of the version"?

Reply:
The latter (objects are always supported regardless of the version). This gets called indeed, e.g. when you have an object below an object.